File: Programmers/[C++]Programmers/#K번째수_240623.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;

    // command 개수만큼 반복 (return 배열 요소 개수)
    for (int i = 0; i < commands.size(); i++) {
        // 자른 숫자 담을 벡터
        vector<int> tmp;
        // i-1부터 j까지 tmp에 담기
        for (int j = commands[i][0] - 1; j < commands[i][1]; j++) {
            tmp.push_back(array[j]);
        }
        // tmp 정렬
        sort(tmp.begin(), tmp.end());
        // k번째 숫자 추가
        answer.push_back(tmp[commands[i][2] - 1]);
    }

    return answer;
}

File: Programmers/[Javascript]Programmers/#K번째수_240623.js
// 2차 풀이
// slice를 사용하여 코드를 줄임
function solution(array, commands) {
    var answer = [];
    
    // return 배열 개수만큼 반복
    for(let i=0; i<commands.length; i++){
        // i,j,k 값 가져오기
        let [start, end, point] = commands[i];
        
        // i-1부터 j까지 자르기
        let tmp = array.slice(start - 1, end);
        
        // tmp 숫자 정렬
        tmp.sort((a,b) => a - b);
        // k번째 숫자 answer에 추가
        answer.push(tmp[point - 1]);
    }
    
    return answer;
}


// 1차 풀이
/*
function solution(array, commands) {
    var answer = [];
    
    // return 배열 개수만큼 반복
    for(let i=0; i<commands.length; i++){
        // 자른 숫자 배열
        let tmp = [];
        
        // i-1부터 j까지 자르기
        for(let j=commands[i][0] - 1; j<commands[i][1]; j++){
            tmp.push(array[j]);
        }
        // tmp 숫자 정렬
        tmp.sort((a,b) => a - b);
        // k번째 숫자 answer에 추가
        answer.push(tmp[commands[i][2] - 1]);
    }
    
    return answer;
}
*/